### DISCUSSION TITLE
Mesh instancing

### DISCUSSION THREAD

**Discussion by ABTN-rtsummit** [notgiven688/jitterphysics2#166] [How do I ... ?]
Hello. I'm looking for neat c# physics engine that can handle a large map, and I found your project.
The map has the same mesh placed with difference scales, rotations, positions.
Is there a way to handle this using single instance?

**Comment by notgiven688** [comment]
Hi. What is a "mesh" in this case?

**Reply by ABTN-rtsummit** [reply]
I'm trying to export  UE5 StaticMeshes in map and convert to triangle meshes in server.

**Reply by notgiven688** [reply]
In Jitter you can add single triangles to the collision system. Each triangles holds a *reference* to a TriangleMesh-instance. In theory you could wrap these triangles using the TransformedShape-shape to give them scale, rotation and position. However this would be less efficient than just creating a single (brute force) TriangleMesh just containing everything. I would suggest to try the brute force way first.

There is an option to write custom collision code which one could use to handle your case. See the "Custom Collision (Octree)" demo. However some additional work is then needed.
